Marlin/planner.cpp

```diff
@@ -983,30 +983,21 @@ void Planner::_buffer_line(const float &a, const float &b, const float &c, const
   const uint8_t moves_queued = movesplanned();
 
   // Slow down when the buffer starts to empty, rather than wait at the corner for a buffer refill
+  unsigned long segment_time = lround(1000000.0 / inverse_mm_s);
   #if ENABLED(SLOWDOWN)
     // Segment time im micro seconds
-    unsigned long segment_time = lround(1000000.0 / inverse_mm_s);
     if (moves_queued > 1 && moves_queued < (BLOCK_BUFFER_SIZE) / 2) {
       if (segment_time < min_segment_time) {
         // buffer is draining, add extra time.  The amount of time added increases if the buffer is still emptied more.
         inverse_mm_s = 1000000.0 / (segment_time + lround(2 * (min_segment_time - segment_time) / moves_queued));
-        #if defined(XY_FREQUENCY_LIMIT) || ENABLED(ENSURE_SMOOTH_MOVES)
```

Comment:
`segment_time` is calculated anyway in (new) line 986.
When SLOWDOWN is defined `inverse_mm_s` is recalculated.
When additionally `XY_FREQUENCY_LIMIT` is defined `segment_time` is updated with the new value of `inverse_mm_s`.

Looks ok to me. But will try. At the moment SLOWDOWN seems to work worse then yesterday - so might be the problem.